lib/frontend/src/ssa.rs

```diff
@@ -104,17 +104,28 @@ impl<Variable> BlockData<Variable> {
                 // in all non-pathological cases
                 let pred: usize = data.predecessors
                     .iter()
-                    .position(|pair| pair.1 == inst)
+                    .position(|&PredBlock { branch, .. }| branch == inst)
                     .expect("the predecessor you are trying to remove is not declared");
-                data.predecessors.swap_remove(pred).0
+                data.predecessors.swap_remove(pred).block
             }
         }
     }
 }
 
+pub struct PredBlock {
```

Comment:
This is local to this file, so it doesn't need to be `pub`.

Reply:
Can't actually be done! The public fn `predecessors` returns a slice of `PredBlock`, so this must be public too.

Reply:
Ah, you're right. It's used in frontend.rs, which doesn't really use `PredBlock`, it just tests if there are *any* predecessors. What if we add a `has_any_predecessors()` function in ssa.rs? Then `predecessors()` could be private too.